packages/@aws-cdk/aws-cloudwatch/lib/insights-rule-body.ts

```diff
@@ -44,256 +44,208 @@ export enum InsightsRuleBodyFilterOperations {
-      filterOperand: this.filterOperand,
-      filterIgnoreCase: this.filterIgnoreCase,
-      filterStatistic: this.filterStatistic,
-    });
   }
+}
 
-  /**
-     * Currently, all string[] operands can only have a max length of 10. This checks if that is satisfied,
-     * otherwise, it will error out
-     * @param operand the value the match is compared against in the filter
-     * @private
-     */
-  private checkAndErrorIfBadOperand(operand: string[]) {
-    if (operand.length > this.MAX_OPERAND_ARR_LEN || operand.length < this.MIN_OPERAND_ARR_LEN) {
-      throw new Error(`Operand array has a max length of ${this.MAX_OPERAND_ARR_LEN} and a minimum length of `+
-        `${this.MAX_OPERAND_ARR_LEN} but given operand length was ${operand.length}`);
-    }
-  }
+/**
+ * A public API to use the filter operation functions to add filter operations to a filter.
+ *
+ * They are entirely the same as the base class, so nothing is added.
+ */
+export class CloudWatchLogsV1FilterOperationFunctions extends InsightsRuleBodyFilterOperationFunctions {
+}
 
+/**
+ * One day the import classes may have something in common, so this is a parent class that will define those
+ * common attributes and/or methods
+ */
+class InsightsRuleBodyFilter {
 }
 
 /**
- * Class that defines the static public APIs for adding filters to a rule body
+ * Class that provides static APIs to add filters to a cloudwatch log v1 rule body
  */
-export class InsightsRuleBodyFilter {
+export class CloudWatchLogsV1Filter extends InsightsRuleBodyFilter {
 
   /**
-     * Converts a IInsightsRuleBodyFilter interface to filter JSON to be used in a rule body
-     * @param filter the filter interface one wishes to use in a rule body
+   * Converts an ICloudWatchLogV1RuleBodyFilter to filter JSON to be used in a rule body
+   * @param filter the filter interface one wishes to use in a rule body
+   */
+  public static fromFilter(filter: ICloudWatchLogV1RuleBodyFilter): any {
+    /**
+     *  InsightsRuleBodyFilterOperationAndInput is defined such that it can only have 1 key and 1 value. Furthermore,
+     *  since it is a "type" the user cannot add more fields to it, so assuming the operation and operand
+     *  are at index 0 are fine
      */
-  public static fromFilter(filter: IInsightsRuleBodyFilter): any {
-    return dropUndefined({
-      Match: filter.filterMatch,
-      [filter.filterOperation]: filter.filterOperand,
-      IgnoreCase: filter.filterIgnoreCase,
-      Statistic: filter.filterStatistic,
-    });
+    return {
+      Match: filter.match,
+      [Object.keys(filter.operationAndInput)[0]]: Object.values(filter.operationAndInput)[0],
+    };
   }
 
+  /**
+   * Converts multiple filters (either given in JSON or in interface form) to filter JSON to be used in a rule body
+   * @param filters the filter interfaces and/or filter JSON one wished to use in a rule body
+   */
+  public static allOf(...filters: ICloudWatchLogV1RuleBodyFilter[]): any[] {
```

Comment:
Is this naming convention in line with CDK guidelines? Did you consider using `fromFilters`?

Reply:
This is a good point. This is my logic for why it should be this way:

When multiple filters are given, the end result is calculated through a logical AND. In the composite alarm library, for defining alarm rules, one can choose multiple alarms and combine them using a logical AND using 'allOf' (similarly, they can combine them using a logical OR using 'anyOf'). This all said, since the filters are combined in a similar fashion (logical AND) and the composite alarms are in the same package, I thought it would be best to name them similarly. Furthermore, as Jeffery said in the design review, there is a possibility that filters can one day be combined via a logical OR, so when that combines one can just make a function 'anyOf' that is like the one in composite alarms.